core/types/access_witness.go

```diff
@@ -125,18 +126,15 @@ func (aw *AccessWitness) touchAddress(addr []byte, isWrite bool) (bool, bool, bo
 		aw.Branches[stem] = AccessWitnessReadFlag
 	}
 
-	selectorRead = true
-
 	// Check for the presence of the leaf selector
 	if _, hasSelector := aw.Chunks[common.BytesToHash(addr)]; !hasSelector {
+		selectorRead = true
```

Comment:
Every `touch` is a read regardless of whether the key is in the `AccessWitness`.  Tangential note from the spec: "Note that when a write takes place, an access event also takes place"